src/test/java/com/google/maps/GeocodingApiTest.java

```diff
@@ -36,60 +27,86 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 
-@Category(LargeTests.class)
-public class GeocodingApiTest extends AuthenticatedTest {
+import static com.google.maps.model.ComponentFilter.administrativeArea;
+import static com.google.maps.model.ComponentFilter.country;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static com.google.maps.TestUtils.retrieveBody;
+
 
-  private static final double EPSILON = 0.005;
+@Category(MediumTests.class)
+public class GeocodingApiTest {
 
-  private GeoApiContext context;
+  private static final double EPSILON = 0.005;
+  private static String simpleGeocodeResponse;
+  private static String placeGeocodeResponse;
+  private static String reverseGeocodeResponse;
+  private static String simpleReverseGeocodeResponse;
+  private static String utfResultGeocodeResponse;
+  private static String reverseGeocodeWithKitaWardResponse;
 
-  public GeocodingApiTest(GeoApiContext context) {
-    this.context = context
-        .setQueryRateLimit(3)
-        .setConnectTimeout(1, TimeUnit.SECONDS)
-        .setReadTimeout(1, TimeUnit.SECONDS)
-        .setWriteTimeout(1, TimeUnit.SECONDS);
+  public GeocodingApiTest() {
+    simpleGeocodeResponse = retrieveBody("SimpleGeocodeResponse.json");
+    placeGeocodeResponse = retrieveBody("PlaceGeocodeResponse.json");
+    reverseGeocodeResponse = retrieveBody("ReverseGeocodeResponse.json");
+    simpleReverseGeocodeResponse = retrieveBody("SimpleReverseGeocodeResponse.json");
+    utfResultGeocodeResponse = retrieveBody("UtfResultGeocodeResponse.json");
+    reverseGeocodeWithKitaWardResponse = retrieveBody("ReverseGeocodeWithKitaWardResponse.json");
   }
 
   @Test
   public void testSimpleGeocode() throws Exception {
-    GeocodingResult[] results = GeocodingApi.newRequest(context).address("Sydney").await();
-    checkSydneyResult(results);
+    try(LocalTestServerContext sc = new LocalTestServerContext(simpleGeocodeResponse)) {
+      GeocodingResult[] results = GeocodingApi.newRequest(sc.context).address("Sydney").await();
+      checkSydneyResult(results);
+
+      sc.assertParamValue("Sydney", "address");
+    }
   }
 
   @Test
   public void testPlaceGeocode() throws Exception {
-    GeocodingResult[] results = GeocodingApi.newRequest(context)
-        .place("ChIJP3Sa8ziYEmsRUKgyFmh9AQM")
-        .await();
-    checkSydneyResult(results);
+    try(LocalTestServerContext sc = new LocalTestServerContext(placeGeocodeResponse)) {
+      String placeID = "ChIJP3Sa8ziYEmsRUKgyFmh9AQM";
+      GeocodingResult[] results = GeocodingApi.newRequest(sc.context)
+          .place(placeID)
+          .await();
+      checkSydneyResult(results);
+
+      sc.assertParamValue(placeID, "place_id");
+    }
   }
 
   @Test
   public void testAsync() throws Exception {
```

Comment:
This is a weird test to have in GeocodingApiTests - maybe it belongs it a context or pending result test? Not particularly relevant to your change though so feel free to ignore.